src/common/proc.c

```diff
@@ -177,7 +177,8 @@ crt_proc_daos_prop_t(crt_proc_t proc, crt_proc_op_t proc_op, daos_prop_t **data)
 		prop->dpp_reserv = tmp;
 		rc = crt_proc_prop_entries(proc, proc_op, prop);
 		if (rc) {
-			daos_prop_free(prop);
+			D_FREE(prop->dpp_entries);
+			D_FREE(prop);
```

Comment:
This change fixes the crash and will want backporting.

Reply:
@ashleypittman I guess I am not totally sure why this fixes it either... I am not really familiar with this portion of the codebase either. Maybe someone who has worked in this part of the code would know better?

Reply:
@dmiter @liuxuezhao any ideas?  Without this change there's a crash on free, but this change avoids the issue.

Reply:
@ashleypittman maybe there is some difference in daos_prop_free vs D_FREE? I am having trouble finding the implementation of daos_prop_free.

Reply:
https://github.com/daos-stack/daos/blob/master/src/common/prop.c#L94

The error that this changes fixes is this one, https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5655/8/valgrindResult/pid=61934,0x0/ bit I can't see how dpe_val_ptr would ever be set.